### PR TITLE
(fix) O3-1973: Separate active orders and past orders in the medications  summary table

### PR DESCRIPTION
**Pull request by Jexsie**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

 - Gets all orders and filters out **pastOrders** and **activeOrders** to solve the issue explained in the jira link below

### Before
Screenshot from 2023-03-20 11-02-34

### After
image

## Related Issue

https://issues.openmrs.org/browse/O3-1973

### CODE REVIEW

**Comment by Jexsie** [comment]
cc @ibacher

**Review by ayush-AI — COMMENTED** [review]
Thanks for the good work @Jexsie

File: packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx [hunk 1]
```diff
@@ -16,36 +16,41 @@ export default function MedicationsSummary({ patientUuid }: MedicationsSummaryPr
...
+  const pastOrders = allOrders?.filter((order) => {
+    if (order.dateStopped) {
+      return order;
+    }
+
+    if (order.autoExpireDate && order.autoExpireDate < currentDate) {
+      return order;
+    }
+  });
+  const activeOrders = allOrders?.filter((order) => !pastOrders.includes(order));
```
Comment (ibacher) [inline packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx:37]: Thanks for this!

I'd wrap all of this in `useMemo`, e.g.,

```ts
const [pastOrders, activeOrders] = useMemo(() => {
  const currentDate = new Date();  // move to local scope
	// logic here
}, [allOrders]);
```

Also, running two filters on the same list plus the `includes` means this algorithm is something like O(n^3). The solution with a simple for loop and two arrays where each item gets pushed to either of the two arrays would be O(n).

Finally, check the logic around `order.dateStopped`. In particular, there's no guarantee that `dateStopped`, if it exists, is always in the past.

Comment (Jexsie) [reply packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx]: Concerning the `order.dateStopped`, whenever I do `order.dateStopped && order.dateStopped < currentDate`. 
Past orders show up in the active orders table. 
Screenshot from 2023-03-21 22-34-43

**Review by vasharma05 — APPROVED** [review]
Thanks for the work!

File: packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx [hunk 2]
```diff
@@ -18,34 +19,45 @@ export default function MedicationsSummary({ patientUuid }: MedicationsSummaryPr
...
-    isLoading: isLoadingPastOrders,
-    isValidating: isValidatingPastOrders,
+    data: allOrders,
+    error: error,
+    isLoading: isLoading,
+    isValidating: isValidating,
   } = usePatientOrders(patientUuid, 'any');
 
+  const [pastOrders, activeOrders] = useMemo(() => {
+    const currentDate = parseDate(formatDatetime(new Date()));
```
Comment (vasharma05) [inline packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx:29]: Add the condition for handling all orders as null.

**Review by hadijahkyampeire — APPROVED** [review]
Thanks @Jexsie for this. cc @gracepotma

**Comment by ibacher** [comment]
@hadijahkyampeire This needed a little more work...

**Comment by hadijahkyampeire** [comment]
oh sorry, I didn't know that we discussed it on the coffee break and we thought everything was addressed. I am so sorry @ibacher  I missed that.

**Comment by hadijahkyampeire** [comment]
Should I create a follow up ticket to address the remaining work?

**Comment by gracepotma** [comment]
> Should I create a follow up ticket to address the remaining work?

Sounds great Hadijah, thank you!!

File: packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx (hunk 1 above)
Comment (Jexsie) [reply packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx]: Ok @gracepotma, Ian talked about creating tickets for this, maybe I will up from there.

**Pull request merged**